blog/2022-10-17-closure-objects.md

```diff
@@ -0,0 +1,192 @@
+---
+slug: closure-objects
+title: How to remove Java from JavaScript
+authors: [sgranat]
+image: '/img/blog/closure-objects/js-without-java.png'
+description: Replacing JavaScript classes with closure and simple JS constructions. Write React-like code even without React.
+tags: [JavaScript, class, closure, React, TypeScript]
+---
+
+As a React developer, I have an suspicion to JS classes usage. In React bubble, it feels like a bad practice when I see them somewhere.
+
+However, recently I got a task to rewrite our `@tolgee/core`, which is VanillaJS, and so far, there have been classes everywhere. I somehow felt it is not right. But then I realized that I don't actually know how to replace classes outside of React. Hmm ...
+
+<!--truncate-->
+
+## Wait, what's wrong with the classes?
+
+Classes are basically added syntax sugar which gives you the impression that you are working with a class based programming language when you are actually not. Classes in JavaScript are implemented differently from the others. My main complaints are these:
+
+- How `this` works is a cause of many issues, and correctly understanding it is not easy
+- Inheritance is becoming an outdated concept, and classes encourage you to use it
+- Private fields in classes are not supported everywhere yet, and their syntax is weird
```

Comment:
and typescript private fields are not really private

Reply:
👍